lib/internal/repl.js

```diff
@@ -153,16 +149,31 @@ function setupHistory(repl, historyPath, oldHistoryPath, ready) {
         // Only attempt to use the history if there was any.
         if (oldReplJSONHistory) repl.history = JSON.parse(oldReplJSONHistory);
 
-        if (!Array.isArray(repl.history)) {
-          throw new errors.TypeError('ERR_INVALID_ARG_TYPE',
-                                     typeof repl.history, 'Array');
+        if (Array.isArray(repl.history)) {
+          repl.history = repl.history.slice(0, repl.historySize);
+        } else {
+          threw = true;
+          _writeToOutput(
+            repl,
+            '\nError: The old history file data has to be an Array.\n' +
+              'REPL session history will not be persisted.\n');
         }
-        repl.history = repl.history.slice(0, repl.historySize);
       } catch (err) {
-        if (err.code !== 'ENOENT') {
-          return ready(
-            new errors.Error('ERR_PARSE_HISTORY_DATA', oldHistoryPath));
-        }
+        // Cannot open or parse history file.
+        // Don't crash, just don't persist history.
+        threw = true;
+        const type = err instanceof SyntaxError ? 'parse' : 'open';
+        _writeToOutput(repl, `\nError: Could not ${type} old history file.\n` +
+                            'REPL session history will not be persisted.\n');
+      }
+      if (!threw) {
+        // Grab data from the older pre-v3.0 JSON NODE_REPL_HISTORY_FILE format.
+        _writeToOutput(
+          repl,
+          '\nConverting old JSON repl history to line-separated history.\n' +
```

Comment:
This was intentional as the message would otherwise be visible even though the conversion did not occur due to e.g. `ENOENT` or a parsing error etc.